Code/Framework/AzCore/AzCore/Name/NameDictionary.cpp

```diff
@@ -106,19 +106,21 @@ namespace AZ
 
         [[maybe_unused]] bool leaksDetected = false;
 
-        for (const auto& keyValue : m_dictionary)
+        for (auto i = m_dictionary.begin(), last = m_dictionary.end(); i != last;)
```

Comment:
huh, I thought one of the benefits of iterators was that you can do the origin for loop

Reply:
oh I see, you need to remove the entry